snapcraft/internal/mangling.py

```diff
@@ -112,26 +114,25 @@ def handle_glibc_mismatch(*, elf_files: FrozenSet[elf.ElfFile],
                    'files and the linker version (2.23) used in the '
                    'base. These are the GLIBC versions required by '
                    'the primed files that do not match and will be '
-                   'patched:\n {}\n'
-                   'To work around this, the newer libc will be '
-                   'migrated into the snap, and these files will be '
-                   'patched to use it.'.format('\n'.join(formatted_list)))
+                   'patched:\n{}\n'.format('\n'.join(formatted_list)))
```

Comment:
Once folks hit the "Hey! Stage libc!" error, does this warning add any meaning? Would it be better to show (essentially) this information only if libc isn't staged, and just trudge along and patch if they've staged it?

Reply:
I think you have the process backwards; first we check for libc6 and only then do we list what needs to be patched.

Reply:
I do think it is valuable, it is essentially modifying binaries that would not be modified otherwise

Reply:
The order doesn't matter, I just question the value-added of this warning once they're already said "oh right right, stage libc". You're right though, this is modifying binaries, we should probably continue to show it.